relations/jf/src/note.rs

```diff
@@ -0,0 +1,167 @@
+use ark_ff::Zero;
+use jf_primitives::circuit::rescue::RescueNativeGadget;
+use jf_relation::{Circuit, PlonkCircuit, Variable};
+
+use crate::{
+    shielder_types::{convert_hash, Note, Nullifier, TokenAmount, TokenId, Trapdoor},
+    CircuitField, PlonkResult, PublicInput,
+};
+
+pub enum NoteType {
+    Deposit,
+    Spend,
+}
+
+pub struct SourcedNote {
+    pub note: Note,
+    pub token_id: TokenId,
+    pub token_amount: TokenAmount,
+    pub trapdoor: Trapdoor,
+    pub nullifier: Nullifier,
+    pub note_type: NoteType,
+}
+
+pub struct SourcedNoteVar {
+    pub note_var: Variable,
+    pub token_id_var: Variable,
+    pub token_amount_var: Variable,
+    pub trapdoor_var: Variable,
+    pub nullifier_var: Variable,
+}
+
+pub trait NoteGadget {
+    fn create_note_variable(&mut self, note: &SourcedNote) -> PlonkResult<SourcedNoteVar>;
+    fn enforce_note_preimage(&mut self, note_var: SourcedNoteVar) -> PlonkResult<Variable>;
+}
```

Comment:
These will eventually be public traits and structs so will need documentation.

Reply:
100% agree, but I wouldn't spend too much time now for docs, while the API is changing so rapidly - for now I would leave things to be clear at least for people working with the code and reviewers